scapy/contrib/loraphy2wan.py

```diff
@@ -0,0 +1,681 @@
+                   BitField("RejoinType", 0, 3),
+                   BitField("DR", 0, 4)]
+
+
+class RejoinParamSetupReq(Packet):
+    name = "RejoinParamSetupReq"
+    fields_desc = [BitField("MaxTimeN", 0, 4),
+                   BitField("MaxCountN", 0, 4)]
+
+
+class RejoinParamSetupAns(Packet):
+    name = "RejoinParamSetupAns"
+    fields_desc = [BitField("RFU", 0, 7),
+                   BitField("TimeOK", 0, 1)]
+
+
+# End of specific 1.1 commands
+
+
+class MACCommand_up(Packet):
+    name = "MACCommand_up"
+    fields_desc = [ByteEnumField("CID", 0, CIDs_up),
+                   ConditionalField(PacketListField("Reset", b"",
+                                                    ResetInd,
+                                                    length_from=lambda pkt:1),
+                                    lambda pkt:(pkt.CID == 0x01)),
+                   ConditionalField(PacketListField("LinkCheck", b"",
+                                                    LinkCheckReq,
+                                                    length_from=lambda pkt:0),
+                                    lambda pkt:(pkt.CID == 0x02)),
+                   ConditionalField(PacketListField("LinkADR", b"",
+                                                    LinkADRReq,
+                                                    length_from=lambda pkt:4),
+                                    lambda pkt:(pkt.CID == 0x03)),
+                   ConditionalField(PacketListField("DutyCycle", b"",
+                                                    DutyCycleReq,
+                                                    length_from=lambda pkt:4),
+                                    lambda pkt:(pkt.CID == 0x04)),
+                   ConditionalField(PacketListField("RXParamSetup", b"",
+                                                    RXParamSetupReq,
+                                                    length_from=lambda pkt:4),
+                                    lambda pkt:(pkt.CID == 0x05)),
+                   ConditionalField(PacketListField("DevStatus", b"",
+                                                    DevStatusReq,
+                                                    length_from=lambda pkt:2),
+                                    lambda pkt:(pkt.CID == 0x06)),
+                   ConditionalField(PacketListField("NewChannel", b"",
+                                                    NewChannelReq,
+                                                    length_from=lambda pkt:5),
+                                    lambda pkt:(pkt.CID == 0x07)),
+                   ConditionalField(PacketListField("RXTimingSetup", b"",
+                                                    RXTimingSetupReq,
+                                                    length_from=lambda pkt:1),
+                                    lambda pkt:(pkt.CID == 0x08)),
+                   ConditionalField(PacketListField("TxParamSetup", b"",  # specific to 1.1 from here  # noqa: E501
```

Comment:
Please avoid using `noqa` tags. Here you can move your comment to another line.

Reply:
done